dev.sh

```
@@ -1,69 +1,200 @@
 API_PORT=9101
 MCP_PORT=9102
 
-echo "🔍 Checking for processes using ports $API_PORT and $MCP_PORT..."
+# Function to rotate logs if they exceed size limit
+rotate_logs() {
+    local log_file=$1
+    local max_size_mb=${2:-10}
+    local max_files=${3:-3}
+    
+    if [ -f "$log_file" ]; then
+        local size_mb=$(du -m "$log_file" | cut -f1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)